Raise bound: QuickCheck

JackKelly-Bellroy opened the pull request:
Tested with:
* `cabal test --constraint 'QuickCheck ^>=2.16'`. 
* `cabal test --constraint 'QuickCheck ^>=2.17' --allow-newer=aeson:QuickCheck `

Could we please have a metadata revision also?

Fixes #112

JackKelly-Bellroy committed:
Raise bound: QuickCheck

ysangkok commented:
Now QuickCheck 2.17 is released, so it probably would make sense to raise the bound to include that version too.

JackKelly-Bellroy commented:
We can propose a bound-raising here, but we can't solve without `allow-newer` because `aeson` won't raise bounds just yet: https://github.com/haskell/aeson/issues/1147#issuecomment-3709877903

Since tests passed with `cabal test --constraint 'QuickCheck ^>=2.17' --allow-newer=aeson:QuickCheck`, I have rebased and updated this commit.

Tristano8 commented:
Pinging this - can it be merged?

thomasjm commented:
Hello, another ping here. Would love to see this released.
